codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestJsonProtocolGenerator.java

```diff
@@ -1064,10 +1082,13 @@ private void generateDocumentBindingStructureShapeDeserializer(
             Predicate<MemberShape> filterMemberShapes
     ) {
         writeJsonTokenizerStartStub(writer, shape);
+        writer.openBlock("if *v == nil {", "}", () -> {
+            writer.write("*v = &$T{}", symbolProvider.toSymbol(shape));
+        });
```

Comment:
Should this always happen regardless if the underlying `*T` is nil?  This will ensure that deserializing an empty object will overwrite any existing values in T that may be present. Kind of contrived given the context of deserialization. But I'd expect it to be more of an error if T already had a value.